2-advanced/dubbo-samples-service-discovery/dubbo-servicediscovery-migration/dubbo-servicediscovery-migration-provider2/src/main/java/org/apache/dubbo/demo/provider/Application.java

```diff
@@ -16,16 +16,15 @@
  */
 package org.apache.dubbo.demo.provider;
 
-import org.springframework.context.support.ClassPathXmlApplicationContext;
-
-import java.util.concurrent.CountDownLatch;
+import org.apache.dubbo.config.spring.context.annotation.EnableDubbo;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
 
+@EnableDubbo
+@SpringBootApplication
 public class Application {
-    public static void main(String[] args) throws Exception {
-        ClassPathXmlApplicationContext context = new ClassPathXmlApplicationContext("spring/dubbo-provider.xml");
-        context.start();
-
-        System.out.println("dubbo service started");
-        new CountDownLatch(1).await();
+    public static void main(String[] args) {
+        new EmbeddedZooKeeper(2181,false).start();
```

Comment:
Will this EmbeddedZooKeeper cause port conflict when running together with `case-configuration.yml`? Because in theory the integration test will also try to launch a zookeeper instance on 2181 port.

Reply:
No, it won't. It's been through integration testing